run_unittests.py

```diff
@@ -2746,6 +2753,39 @@ def test_buildtype_setting(self):
         self.assertEqual(opts['debug'], True)
         self.assertEqual(opts['optimization'], '0')
 
+    def test_introspect_projectinfo_without_configured_build(self):
+        testfile = os.path.join(self.common_test_dir, '36 run program', 'meson.build')
+        res = self.introspect_directory(testfile, '--projectinfo')
+        self.assertEqual(res, {'buildsystem_files': ['meson.build'],
+                               'name': 'run command',
+                               'version': None,
+                               'descriptive_name': 'run command',
+                               'subprojects': []
+                              })
+
+        testfile = os.path.join(self.common_test_dir, '44 options', 'meson.build')
+        res = self.introspect_directory(testfile, '--projectinfo')
+        self.assertEqual(res, {'buildsystem_files': ['meson_options.txt', 'meson.build'],
+                               'name': 'options',
+                               'version': None,
+                               'descriptive_name': 'options',
+                               'subprojects': []
+                              })
+
+        testfile = os.path.join(self.common_test_dir, '47 subproject options', 'meson.build')
+        res = self.introspect_directory(testfile, '--projectinfo')
+        self.assertEqual(res, {'buildsystem_files': ['meson_options.txt', 'meson.build'],
+                               'name': 'suboptions',
+                               'version': None,
+                               'descriptive_name': 'suboptions',
+                               'subprojects': [
+                                   {'buildsystem_files': ['subprojects/subproject/meson_options.txt', 'subprojects/subproject/meson.build'],
+                                    'name': 'subproject',
+                                    'version': 'undefined',
+                                    'descriptive_name': 'subproject'}
+                                   ]
```

Comment:
__[Flake8]__

[E123] closing bracket does not match indentation of opening bracket's line